internal/service/ipam/api_address_block_resource_test.go

```diff
@@ -0,0 +1,1126 @@
+					resource.TestCheckResourceAttrPair(resourceName, "space", "bloxone_ipam_ip_space.two", "id"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAddressBlockResource_Tags(t *testing.T) {
+	var resourceName = "bloxone_ipam_address_block.test_tags"
+	var v ipam.IpamsvcAddressBlock
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAddressBlockTags("192.168.0.0", "16", map[string]string{"location": "NA"}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.location", "NA"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAddressBlockTags("192.168.0.0", "16", map[string]string{"location": "CA"}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAddressBlockExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.location", "CA"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccCheckAddressBlockExists(ctx context.Context, resourceName string, v *ipam.IpamsvcAddressBlock) resource.TestCheckFunc {
+	// Verify the resource exists in the cloud
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		apiRes, _, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			AddressBlockAPI.
+			AddressBlockRead(ctx, rs.Primary.ID).
+			Execute()
+		if err != nil {
+			return err
+		}
+		if !apiRes.HasResult() {
+			return fmt.Errorf("expected result to be returned: %s", resourceName)
+		}
+		*v = apiRes.GetResult()
+		return nil
+	}
+}
+
+func testAccCheckAddressBlockDestroy(ctx context.Context, v *ipam.IpamsvcAddressBlock) resource.TestCheckFunc {
+	// Verify the resource was destroyed
+	return func(state *terraform.State) error {
+		_, httpRes, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			AddressBlockAPI.
+			AddressBlockRead(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+				// resource was deleted
+				return nil
+			}
+			return err
+		}
+		return errors.New("expected to be deleted")
+	}
+}
+
+func testAccCheckAddressBlockDisappears(ctx context.Context, v *ipam.IpamsvcAddressBlock) resource.TestCheckFunc {
+	// Delete the resource externally to verify disappears test
+	return func(state *terraform.State) error {
+		_, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			AddressBlockAPI.
+			AddressBlockDelete(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func testAccAddressBlockIPSpace() string {
```

Comment:
You can use `testAccBaseWithIPSpace` instead, and delete this duplicate function

https://github.com/infobloxopen/terraform-provider-bloxone/blob/35d0e1d2ace37835ce215edd00b8371606f41499/internal/service/ipam/api_subnet_resource_test.go#L884-L891